lib/services/plugins-service.ts

```diff
@@ -36,9 +36,24 @@ export class PluginsService implements IPluginsService {
 	public add(plugin: string): IFuture<void> {
 		return (() => {
 			this.ensure().wait();
-			let dependencyData = this.$npm.cache(plugin, undefined, PluginsService.NPM_CONFIG).wait();
-			if (dependencyData.nativescript) {
-				let pluginData = this.convertToPluginData(dependencyData);
+			let installedModuleNames = this.$npm.install(plugin, this.$projectData.projectDir, PluginsService.NPM_CONFIG);
+
+			//npmInstallResultStream is not very usefull (doesn't contain all the package information), so we need to find the real one inside node_modules
+			let foundValidPlugin = false;
+			let realNpmPackageJson:any;
+			for(let index in installedModuleNames) {
```

Comment:
you can use `of` instead of `in` which will give you the name directly:
```TypeScript
for(let name of installedModuleNames) {
...
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of